### PR TITLE
Add support for the new 'site' parameter in the agent config

### PR DESCRIPTION
**Pull request by sfluor**
This adds support for the new `site` parameter in the agent config.

### CODE REVIEW

**Review by shang-wang — APPROVED** [review]

**Review by conorbranagan — APPROVED** [review]

**Comment by conorbranagan** [comment]
Let's get someone from agent core to look just to verify it matches the expectations?

**Review by olivielpeau — COMMENTED** [review]
For reference, related PR on `datadog-agent`: https://github.com/DataDog/datadog-agent/pull/2412

I haven't looked at the details, but since the process-agent already uses the `config` package exposed by `datadog-agent`, you could probably use it directly, I'm guessing that would reduce the amount of custom logic required on process-agent's side. It would require the changes in the `datadog-agent` PR I linked to though. What do you think?

**Comment by sfluor** [comment]
I think it's a better idea yes 👍 , I will look into it

File: config/config.go
Comment (sfluor) [reply config/config.go]: Thanks for pointing this out 👍  I will remove the `else if ` statement, however the first `if` statement (line 463) will be needed I think because the env name for the `process_dd_url` is not handled by the datadog-agent config package (not the same convention): https://github.com/DataDog/datadog-agent/pull/2412#discussion_r223436757

Comment (conorbranagan) [reply config/config.go]: Oh yeah, correct

**Review by conorbranagan — COMMENTED** [review]
This looks good to merge once the dependent PR is merged

**Review by olivielpeau — APPROVED** [review]
Looks good! Dependent `datadog-agent` PR has been merged, this PR should only need an update on the revision of `datadog-agent`.

**Pull request merged**
